src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,1473 @@
+        when(oldCartDiscount.getTarget())
+                .thenReturn(MultiBuyCustomLineItemsTarget.of("quantity > 0", 6L, 2L,  SelectionMode.MOST_EXPENSIVE));
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final MultiBuyCustomLineItemsTarget newTarget =
+                MultiBuyCustomLineItemsTarget.of("quantity > 0", 6L, 3L, SelectionMode.MOST_EXPENSIVE);
+        when(newCartDiscountDraft.getTarget()).thenReturn(newTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+                buildChangeTargetUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).contains(ChangeTarget.of(newTarget));
+    }
+
+    @Test
+    void buildChangeTargetUpdateAction_WithDifferentCustomSelectionMode_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getTarget())
+                .thenReturn(MultiBuyCustomLineItemsTarget.of("quantity > 0", 6L, 3L,  SelectionMode.CHEAPEST));
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final MultiBuyCustomLineItemsTarget newTarget =
+                MultiBuyCustomLineItemsTarget.of("quantity > 0", 6L, 3L, SelectionMode.MOST_EXPENSIVE);
+        when(newCartDiscountDraft.getTarget()).thenReturn(newTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+                buildChangeTargetUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).contains(ChangeTarget.of(newTarget));
+    }
+
+    @Test
+    void buildChangeTargetUpdateAction_WithDifferentCustomMaxOccurrence_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getTarget())
+                .thenReturn(MultiBuyCustomLineItemsTarget
+                        .of("quantity > 0", 6L, 3L,  SelectionMode.MOST_EXPENSIVE, 1L));
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final MultiBuyCustomLineItemsTarget newTarget =
+                MultiBuyCustomLineItemsTarget.of("quantity > 0", 6L, 3L, SelectionMode.MOST_EXPENSIVE, 2L);
+        when(newCartDiscountDraft.getTarget()).thenReturn(newTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+                buildChangeTargetUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).contains(ChangeTarget.of(newTarget));
+    }
+
+    @Test
+    void buildChangeTargetUpdateAction_WithSameMultiBuyCustomLineItemsTargetValues_ShouldNotBuildUpdateAction() {
+        // Given a cart with 6 items, the discount can be applied only once.
+        // As a result, 3 custom items will be discounted
+        // and 4 most expensive custom items will be marked as participating in this discount.
```

Comment:
I think these comments are confusing to be here. They don't affect the tests in any way.

That's other than the fact that they are wrong 🙈

Reply:
Please add a test for testing same values of different instances (which would be the more common case actually happening)